contracts/AstarBase.sol

```diff
@@ -0,0 +1,146 @@
+        require(!paused, "The unRegistration is paused");
+        require(addressMap[msg.sender] != 0, "Unregistring unknown entry");
+
+        uint256 polkadotAddr = addressMap[msg.sender];
+        addressMap[msg.sender] = 0;
+        polkadotAddrMap[polkadotAddr] = FREE_ADDRESS;
+        registeredCnt.decrement();
+    }
+
+    /// @notice unRegister any address by contract owner
+    /// @param evmAddress, EVM address used for registration
+    function sudoUnRegister(address evmAddress) public onlyOwner {
+        require(addressMap[evmAddress] != 0, "Unregistring unknown entry");
+
+        uint256 polkadotAddr = addressMap[evmAddress];
+        addressMap[evmAddress] = 0;
+        polkadotAddrMap[polkadotAddr] = FREE_ADDRESS;
+        registeredCnt.decrement();
+    }
+
+    /// @notice change registered polkadot address for the given evmAddress
+    /// @param polkadotAddr, SS58 address used for staking
+    function changeRegistered(uint256 polkadotAddr) public {
+        require(!paused, "The changing Registration is paused");
+        require(addressMap[msg.sender] != 0, "Changing unregistered entry");
+        require(polkadotAddr != 0, "Can't changeRegistered polkadotAddr with 0");
+        require(isFreePolkadotAddr(polkadotAddr), "Already used staker's address");
+
+        // free old polkadot address
+        uint256 oldPolkadotAddr = addressMap[msg.sender];
+        polkadotAddrMap[oldPolkadotAddr] = FREE_ADDRESS;
+
+        // store ss58 address used for staking
+        addressMap[msg.sender] = polkadotAddr;
+
+        // use polkadotAddrMap to prevent reuse of same address
+        polkadotAddrMap[polkadotAddr] = msg.sender;
+
+        registeredCnt.decrement();
+    }
+
+    /// @notice Check if given address was registered
+    /// @param evmAddress, EVM address used for registration
+    function isRegistered(address evmAddress) public view returns (bool) {
+        uint256 polkadotAddr = addressMap[evmAddress];
+        if (polkadotAddr == 0) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    /// @notice Check if given address was registered and returned staked amount
+    /// @param evmAddress, EVM address used for registration
+    /// @return Staked amount on the SS58 address
+    function stakedAmount(address evmAddress) public returns (uint128) {
+        require(onChainCheckEnabled == true, "On-chain check is not enabled");
+        if (isRegistered(evmAddress) == true) {
+            // uint256 polkadotAddr = addressMap[evmAddress];
+            return verifyOnChain(evmAddress);
+        }
+        return 0;
+    }
+
+    /// @notice Verify with the dapps-staking precompile how much the SS58 address is staking
+    /// @param evmAddress, EVM address used for registration
+    /// @return Staked amount on the SS58 address, or 0 if stakers status is not confirmed
+    function verifyOnChain(address evmAddress) private returns (uint128) {
```

Comment:
Strange name since the method doesn't verify anything.

Reply:
oddly enough, the actual verification is done in `isRegistered`
https://github.com/AstarNetwork/astarbase/pull/1/files#diff-063a25b7e5a98bf54c143e99c2f2c55093e6cb58ee19f53866653600a2a6aa4aR91

Reply:
this is just temporary for tests to pass